src/sync.h

```diff
@@ -129,7 +127,25 @@ class LOCKABLE AnnotatedMixin : public PARENT
 using RecursiveMutex = AnnotatedMixin<std::recursive_mutex>;
 
 /** Wrapped mutex: supports waiting but not recursive locking */
-typedef AnnotatedMixin<std::mutex> Mutex;
+using Mutex = AnnotatedMixin<std::mutex>;
+
+/** Different type to mark Mutex's outside of a class
+ *
+ * Thread safety analysis can't handle negative assertions about mutex's
+ * with global scope well, so mark them with a separate type, and
+ * eventually move all the mutexes into classes so they are not globally
```

Comment:
nit, "mutexes" here is correct, but s/Mutex's/Mutexes/ on line 130 and s/mutex's/mutexes/ on line 134